src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Win32.cs

```diff
@@ -245,95 +248,108 @@ private void ReadDirectoryChangesCallback(uint errorCode, uint numBytes, AsyncRe
             }
         }
 
-        private unsafe void ParseEventBufferAndNotifyForEach(byte[] buffer)
+        private unsafe void ParseEventBufferAndNotifyForEach(ReadOnlySpan<byte> buffer)
         {
-            Debug.Assert(buffer != null);
-            Debug.Assert(buffer.Length > 0);
+            ReadOnlySpan<char> oldName = ReadOnlySpan<char>.Empty;
 
-            fixed (byte* b = buffer)
+            // Parse each event from the buffer and notify appropriate delegates
+            for (;;)
```

Comment:
Nit: stylistically, we've generally standardized on `while (true)` instead of `for (;;)`